extras/devices/LVFS/LVFS.py

```diff
@@ -0,0 +1,332 @@
+                        higher_ver
+                    )
+                if lower_ver != []:
+                    vers = (
+                        vers
+                        + " - LVFS downgradable version(s): %s"
+                        % ", ".join(lower_ver)
+                    )
+                if vers:
+                    msg = msg + vers
+                    print(msg)
+            else:
+                msg = msg + " - no available firmware on LVFS"
+            if msg:
+                logger.info(msg)
+
+    def upgrade(self):
+        """
+        Upgrade all devices firmware to latest version if available on LVFS
+
+        :returns: if reboot is required
+        :rtype: boolean
+        """
+        logger.info("start upgrading")
+        reboot = False
+        for dev in self.fw_info:
+            dev_name = dev["Name"] if "Name" in dev else dev["DeviceId"]
+            try:
+                latest_ver = dev["Releases"][0]
+            except KeyError:
+                continue
+            if dev["Version"] != latest_ver["Version"]:
+                if "is-upgrade" in latest_ver["Flags"]:
+                    logger.info(
+                        "[%s] try upgrading to %s"
+                        % (dev_name, latest_ver["Version"])
+                    )
+                    dev["targetVersion"] = latest_ver["Version"]
+                    rc, stdout, stderr = self.run_cmd(
+                        "sudo fwupdmgr upgrade %s -y --no-reboot-check"
+                        % dev["DeviceId"],
+                        raise_stderr=False,
+                    )
+                    if rc == 0:
+                        logger.debug(stdout)
+                        reboot = True
+                    else:
+                        logger.info(
+                            "[%s] Failed to upgrade %s"
+                            % (dev_name, latest_ver["Version"])
+                        )
+                        logger.debug(stdout)
+                        logger.debug(stderr)
+                else:
+                    logger.info(
+                        "[%s] unsupported Flags: %s"
+                        % (dev_name, str(latest_ver["Flags"]))
+                    )
+            else:
+                logger.info(
+                    "[%s] already the latest available firmware version"
+                    % dev_name
+                )
+        return reboot
+
+    def downgrade(self):
+        """
+        Downgrade all devices firmware to 2nd new version if available on LVFS
+
+        :returns: if reboot is required
+        :rtype: boolean
+        """
+        logger.info("start downgrading")
+        reboot = False
+        for dev in self.fw_info:
+            dev_name = dev["Name"] if "Name" in dev else dev["DeviceId"]
+            try:
+                prev_ver = dev["Releases"][1]
+            except KeyError:
+                continue
+            except IndexError:
```

Comment:
You can use a tuple to handle multiple exceptions like `except (KeyError, IndexError)`